### PR TITLE
Added interface for getEntrance task metrics

### LINKED ISSUE #2222: [Feature] [1.2.0]Single LinkisEntrance task metrics collection
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-linkis/issues) and found no similar feature requirement.

### Problem Description

Entrance collects metrics such as running tasks, queued tasks, and the number of failed tasks, both overall and at the user-creator level

### Description

_No response_

### Use case

_No response_

### solutions

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### LINKED ISSUE #2257: [Feature][Entrance] Gateway supports forwarding non-execution related requests of Entrance
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-linkis/issues) and found no similar feature requirement.

### Problem Description

Gateway supports forwarding non-execution-related requests of Entrance, and requests related to /etnrance/api/xx support forwarding to Entrance services, and supports specifying redirection to fixed services through the query parameter instance=ip:port
Gateway支持转发Entrance的非执行相关请求,/etnrance/api/xx相关的请求支持转发到Entrance服务，并支持通过查询参数instance=ip:port指定转向固定的服务

### Description

_No response_

### Use case

_No response_

### solutions

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### LINKED ISSUE #2258: [Feature][Entrance] Added feature to clean up running tasks when the Entrance process exits normally
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-linkis/issues) and found no similar feature requirement.

### Problem Description

Added feature When the Entrance process exits normally, clean up the running tasks, and initiate kill of the running tasks by monitoring the exit time of spring
增加特性Entrance进程正常退出时，清理运行中的任务，通过监听spring的退出时间，将运行中的任务发起kill

### Description

_No response_

### Use case

_No response_

### solutions

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by peacewong**
### What is the purpose of the change
close #2222
close #2258
close #2257
### Brief change log
- LinkisEntrance task metrics collection.
- Added feature to clean up running tasks when the Entrance process exits normally.
- Gateway supports forwarding non-execution related requests of Entrance.

### CODE REVIEW

File: linkis-computation-governance/linkis-entrance/src/main/java/org/apache/linkis/entrance/restful/EntranceMetricRestfulApi.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @RequestMapping(path = "/taskinfo", method = RequestMethod.GET)
+    public Message status(
+            HttpServletRequest req,
+            @RequestParam(value = "user", required = false) String user,
+            @RequestParam(value = "creator", required = false) String creator,
+            @RequestParam(value = "engineTypeLabel", required = false)
+                    String engineTypeLabelValue) {
+        String userName = ModuleUserUtils.getOperationUser(req, "taskcount");
```
Comment (legendtkl) [inline linkis-computation-governance/linkis-entrance/src/main/java/org/apache/linkis/entrance/restful/EntranceMetricRestfulApi.java:58]: nit: "taskcount" here looks a little bit confused

File: linkis-computation-governance/linkis-entrance/src/main/java/org/apache/linkis/entrance/restful/EntranceMetricRestfulApi.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        if (!Configuration.isAdmin(userName)) {
+            if (StringUtils.isBlank(queryUser)) {
+                queryUser = userName;
+            } else if (!userName.equalsIgnoreCase(queryUser)) {
+                return Message.error(
+                        "Non-administrators cannot view other users' task information");
+            }
+        }
+        String filterWords = creator;
+        if (StringUtils.isNoneBlank(filterWords) && StringUtils.isNoneBlank(queryUser)) {
```
Comment (legendtkl) [inline linkis-computation-governance/linkis-entrance/src/main/java/org/apache/linkis/entrance/restful/EntranceMetricRestfulApi.java:69]: nit: for single parameter, I think `StringUtils.isNotBlank` is better.

File: linkis-spring-cloud-services/linkis-service-gateway/linkis-gateway-server-support/src/main/scala/org/apache/linkis/gateway/ujes/parser/EntranceRequestGatewayParser.scala [hunk 3]
```diff
@@ -30,12 +32,27 @@ class EntranceRequestGatewayParser extends AbstractGatewayParser {
...
+      }
+      gatewayContext.getGatewayRoute.setServiceInstance(serviceInstance)
     case _ =>
   }
 }
 
 object EntranceRequestGatewayParser {
   val ENTRANCE_REQUEST_REGEX = (ENTRANCE_HEADER + "([^/]+)/.+").r
+  val API_REQUEST = "api"
+  val INSTANCE = "instance"
```
Comment (legendtkl) [inline linkis-spring-cloud-services/linkis-service-gateway/linkis-gateway-server-support/src/main/scala/org/apache/linkis/gateway/ujes/parser/EntranceRequestGatewayParser.scala:57]: An empty line in the end of file is recommanded.

**Review by legendtkl — COMMENTED** [review]
LGTM with some nit comments.

File: linkis-computation-governance/linkis-entrance/src/main/java/org/apache/linkis/entrance/restful/EntranceMetricRestfulApi.java (hunk 2 above)
Comment (peacewong) [reply linkis-computation-governance/linkis-entrance/src/main/java/org/apache/linkis/entrance/restful/EntranceMetricRestfulApi.java:69]: > nit: for single parameter, I think `StringUtils.isNotBlank` is better.

yes

**Pull request merged**